### DISCUSSION TITLE
How to call functions of CLP using Sonnet?

### DISCUSSION THREAD

**Discussion by Smartisa** [coin-or/Sonnet#26] [General]
If I want to use CLP alone to implement a certain function, how can I achieve it?

**Comment by jhmgoossens** [comment]
(Converted the Issue to Discussion)

Can you give an example of what you're trying to do?

Every class and method of the underlying Cbc or Clp (etc.) that you want to use in .NET has to first be individually wrapped in SonnetWrapper. For now, only classes and methods necessary for Sonnet itself have been wrapped in SonnetWrapper. For many methods--especially with a simple signature--adding a wrapper is easy.
Once a class and method is wrapped in SonnetWrapper, you can then use them in .NET code, like in [Sonnet Cbc Tests](https://github.com/coin-or/Sonnet/blob/master/test/SonnetTest/Sonnet_CbcTests.cs#L48).
There is also some non-working [code in SonnetWrapperTest.cpp](https://github.com/coin-or/Sonnet/blob/master/test/SonnetWrapperTest/SonnetWrapperTest.cpp) but it is probably illustrative of what's already possible.
So, it depends on what you're trying to do in CLP?

**Reply by Smartisa** [reply]
Because I saw that CLP performed very well in a test report（http://plato.asu.edu/ftp/lpsimp.html）, I wanted to use CLP to realize some Boolean integer linear programming functions, but .Net project can't call CLP. Seeing sonnet wrap CLP in sonnetwrapper, I want to ask about the use of CLP.
In addition, which of sonnet and CLP has better performance in processing linear programming?

**Reply by jhmgoossens** [reply]
Oh, then you're in luck! Sonnet is not a solver, but more like a modelling language for .NET to then let Cbc or Clp or another solver actually solve the problem. Please look at any of the examples in the [examples folder](https://github.com/coin-or/Sonnet/tree/master/examples) on how to model a problem and then solve it using Cbc or Clp. The example show different programming techniques.

If your problem has integer variables, then you will get much better results using Cbc rather than Clp, since Clp is for linear programming problems with all real-valued variables that are _not_ restricted to be integer or even binary. The benchmark report you mention is for pure LP, so with only real-valued variables, _not_ for integer linear programming.

Anyway, just try to use Sonnet as in the Examples! 
If you have more questions, get back to us here.
